### PR TITLE
Decouple llms.txt diff check from LOC gate

### PR DESCRIPTION
**Pull request by d-o-hub**
Decoupled the `llms.txt` synchronization check from the standard CI linting process. Instead of failing on stale artifacts, CI now enforces a public API surface threshold (5000 symbols) to prevent uncontrolled growth. Automation for updating and committing these artifacts has been moved to the release workflow. Pre-commit hook still regenerates them for local visibility but no longer auto-stages them.

Fixes #175

---
*PR created automatically by Jules for task [3286101680947605025](https://jules.google.com/task/3286101680947605025) started by @d-o-hub*

### CODE REVIEW

File: scripts/validate.sh [hunk 1]
```diff
@@ -63,12 +63,18 @@ fi
...
 scripts/gen-llms-txt.sh
-if ! git diff --quiet llms.txt llms-full.txt 2>/dev/null; then
-  echo "❌ llms.txt or llms-full.txt was modified. Run scripts/gen-llms-txt.sh and commit the changes."
-  git diff llms.txt llms-full.txt
+
+LOC=$(grep -cE '^\s*(pub |fn |struct |enum |trait |impl )' llms-full.txt || true)
+echo "Public API surface: $LOC symbols"
+
+THRESHOLD=5000
+if [ "$LOC" -gt "$THRESHOLD" ]; then
```
Comment (d-o-hub) [inline scripts/validate.sh]: @jules address feedback Use '[[' instead of '[' for conditional tests. The '[[' construct is safer and more feature-rich.
Check failure: Use '[[' instead of '[' for conditional tests. The '[[' construct is safer and more feature-rich.

See more on https://sonarcloud.io/project/issues?id=d-o-hub_chaotic_semantic_memory&issues=AZ3xvIGq5yp55qrRJbtQ&open=AZ3xvIGq5yp55qrRJbtQ&pullRequest=176

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -132,8 +132,13 @@ jobs:
...
 
-    - name: Generate llms.txt files for AI tool integration
-      run: bash scripts/gen-llms-txt.sh
+    - name: Regenerate & commit llms.txt
+      run: |
+        bash scripts/gen-llms-txt.sh
+        git config user.name "github-actions[bot]"
+        git config user.email "github-actions[bot]@users.noreply.github.com"
+        git add llms.txt llms-full.txt
+        git diff --cached --quiet || (git commit -m "chore: regenerate llms.txt for release [skip ci]" && git push)
```
[the remark this answers is not in the document]
Comment (d-o-hub) [reply .github/workflows/release.yml:151]: @jules Address all PR review feedback completely and systematically.

Requirements:

* Analyze every review comment, requested change, CI signal, and inline discussion on the PR.
* Implement fixes with production-quality changes only — no temporary workarounds, commented-out code, or partial resolutions.
* Preserve existing behavior unless explicitly required by the feedback.
* Verify there are no regressions, edge-case failures, or broken integrations.

Validation workflow:

1. Re-run the full validation pipeline locally:

   * tests
   * lint
   * typecheck
   * build
   * benchmarks (if applicable)
2. Use real usage / dogfooding flows where possible instead of only unit tests.
3. Verify affected functionality through the actual CLI and end-to-end execution paths.
4. Confirm CI parity with local execution.
5. Inspect git diff carefully for accidental unrelated changes.

Git workflow:

* Commit all changes with clear conventional commit messages.
* Push all commits to the active PR branch.

PR verification (mandatory):

* Use GitHub CLI (`gh`) to inspect the PR after pushing.
* Verify:

  * all required checks are green or running successfully
  * no merge conflicts exist
  * PR state is `mergeable`
  * no unresolved review threads remain
  * branch is up to date with target branch if required

Use commands such as:

* `gh pr view --comments --json reviews,reviewThreads,mergeable,headRefName,statusCheckRollup`
* `gh pr checks`
* `gh pr status`

If the PR is not mergeable:

* identify the blocking issue
* resolve it completely
* re-verify until the PR is mergeable

Finally provide:

* concise summary of implemented fixes
* validation results
* CI/check status
* final mergeability status
* remaining risks or follow-ups (if any)
